pkg/collector/openid.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type OpenIdConfig struct {
+	Issuer        string `json:"issuer"`
+	JwksUri       string `json:"jwks_uri"`
+	TokenEndpoint string `json:"token_endpoint"`
+	AuthEndpoint  string `json:"authorization_endpoint"`
+}
+
+func validateOpenIdConnectConfig(rawUrl string) error {
+	// append well-known configuration to issuer
+	openIdConnectUrl, err := url.Parse(rawUrl)
+	if err != nil {
+		return err
+	}
```

Comment:
Maybe just accept a valid URL as the input to this function e.g.

```go
validateOpenIdConnectConfig(issuerUrl *url.URL) error {
```

I think the name being more specific than rawUrl could also help the caller know what's expected